File: questPresentation/src/main/java/ru/nekit/android/qls/setupWizard/view/SetupQTPComplexityFragment.java
package ru.nekit.android.qls.setupWizard.view;

import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.RadioGroup;

import ru.nekit.android.qls.R;
import ru.nekit.android.qls.quest.qtp.QuestTrainingProgramComplexity;

public class SetupQTPComplexityFragment extends QuestSetupWizardFragment
        implements RadioGroup.OnCheckedChangeListener {

    private RadioGroup mQTPComplexityGroup;

    public static SetupQTPComplexityFragment getInstance() {
        return new SetupQTPComplexityFragment();
    }

    @Override
    protected void onSetupStart(@NonNull View view) {
        mQTPComplexityGroup = (RadioGroup) view.findViewById(R.id.complexity_group);
        mQTPComplexityGroup.setOnCheckedChangeListener(this);
        setNextButtonText(R.string.label_ok);
        update(false);
    }

    @Override
    protected boolean nextAction() {
        QuestTrainingProgramComplexity complexity = null;
        int selectedComplexity = mQTPComplexityGroup.getCheckedRadioButtonId();
        if (selectedComplexity != -1) {
            if (selectedComplexity == R.id.complexity_easy) {
                complexity = QuestTrainingProgramComplexity.EASY;
            } else if (selectedComplexity == R.id.complexity_normal) {
                complexity = QuestTrainingProgramComplexity.NORMAL;
            } else if (selectedComplexity == R.id.complexity_hard) {
                complexity = QuestTrainingProgramComplexity.HARD;
            }
        }
        if (complexity != null) {
            getSetupWizard().setQTPComplexity(complexity);
        }
        return true;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.sw_setup_qtp_complexity;
    }

    @Override
    public void onCheckedChanged(RadioGroup group, @IdRes int checkedId) {
        update(true);
    }

    private void update(boolean choiced) {
        setNextButtonVisibility(choiced);
    }

    @Override
    public void onDestroy() {
        mQTPComplexityGroup.setOnCheckedChangeListener(null);
        super.onDestroy();
    }
}

File: questPresentation/src/main/java/ru/nekit/android/qls/setupWizard/view/SetupQTPComplexityFragment.kt
package ru.nekit.android.qls.setupWizard.view

import android.support.annotation.LayoutRes
import android.view.View
import android.widget.RadioGroup
import com.jakewharton.rxbinding2.widget.checkedChanges
import io.reactivex.Single
import ru.nekit.android.qls.R
import ru.nekit.android.qls.R.id.*
import ru.nekit.android.qls.shared.model.Complexity.*

class SetupQTPComplexityFragment : QuestSetupWizardFragment() {

    private lateinit var qtpComplexityGroup: RadioGroup

    override fun onSetupStart(view: View) {
        qtpComplexityGroup = view.findViewById(complexity_group)
        autoDispose {
            qtpComplexityGroup.checkedChanges().subscribe {
                update(true)
            }
        }
        setNextButtonText(R.string.label_ok)
        update(false)
    }

    override fun nextAction(): Single<Boolean> {
        return setupWizard.setQTPComplexity(qtpComplexityGroup.checkedRadioButtonId.let {
            when (it) {
                complexity_easy -> EASY
                complexity_normal -> NORMAL
                complexity_hard -> HARD
                else -> null
            }
        })
    }

    @LayoutRes
    override fun getLayoutId() = R.layout.sw_setup_qtp_complexity

    private fun update(choiced: Boolean) {
        setNextButtonVisibility(choiced)
    }

    companion object {

        val instance: SetupQTPComplexityFragment
            get() = SetupQTPComplexityFragment()
    }
}